tests/integration-tests/tests/storage/test_efs.py

```diff
@@ -139,13 +186,22 @@ def _check_efs_after_nodes_reboot(all_mount_dirs, cluster, remote_command_execut
     for compute_node in compute_nodes:
         scheduler_commands.reboot_compute_node(compute_node, asap=False)
     scheduler_commands.wait_nodes_status("idle", compute_nodes)
-    _check_efs_correctly_mounted_and_shared(all_mount_dirs, remote_command_executor, scheduler_commands)
+    _check_efs_correctly_mounted_and_shared(
+        all_mount_dirs, remote_command_executor, scheduler_commands, iam_authorizations, encryption_in_transits
+    )
     return remote_command_executor, scheduler_commands
 
 
-def _check_efs_correctly_mounted_and_shared(all_mount_dirs, remote_command_executor, scheduler_commands):
-    for mount_dir in all_mount_dirs:
-        test_efs_correctly_mounted(remote_command_executor, mount_dir)
+def _check_efs_correctly_mounted_and_shared(
+    all_mount_dirs, remote_command_executor, scheduler_commands, iam_authorizations=None, encryption_in_transits=None
+):
+    for i, mount_dir in enumerate(all_mount_dirs):
+        test_efs_correctly_mounted(
+            remote_command_executor,
+            mount_dir,
+            encryption_in_transits[i] if encryption_in_transits is not None else False,
+            iam_authorizations[i] if iam_authorizations is not None else False,
```

Comment:
Can we remove the conditional statements?

Reply:
No, because it negatively impacts the notion of Don't Repeat Yourself (DRY) and decreases readability.

Reply:
Can you give me an example where these conditional statements are useful?

Reply:
This is no longer relevant given the [explanation](https://github.com/aws/aws-parallelcluster/pull/4668#discussion_r1044769950) for and [removal of the defaults]([remove defaults and ternary checks for iam_authorizations and encrypt…](https://github.com/aws/aws-parallelcluster/pull/4668/commits/8806e33bb4a9a9bfdb765b542465d69e4eb24b57))